File: mall-module/src/main/java/com/fastproject/mall/domain/MallUserAddress.java
package com.fastproject.mall.domain;

import com.fastproject.db.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLRestriction;

/**
 * 商城 - 用户收货地址
 */
@Table(name = "mall_user_address")
@Getter
@Setter
@Entity
@SQLDelete(sql = "update mall_user_address set deleted = 1 where id = ?")
@SQLRestriction("deleted = 0")
public class MallUserAddress extends BaseEntity {

    /**
     * 用户ID
     */
    private Long userId;

    /**
     * 收货人姓名
     */
    private String receiverName;

    /**
     * 收货人电话
     */
    private String receiverPhone;

    /**
     * 省
     */
    private String province;

    /**
     * 省编码
     */
    private String provinceCode;

    /**
     * 市
     */
    private String city;

    /**
     * 市编码
     */
    private String cityCode;

    /**
     * 区/县
     */
    private String district;

    /**
     * 区/县编码
     */
    private String districtCode;

    /**
     * 详细地址
     */
    @Column(length = 500)
    private String detailAddress;

    /**
     * 邮政编码
     */
    private String postalCode;

    /**
     * 是否默认 0 否 1 是
     */
    private Integer isDefault;

    /**
     * 标签（如：家、公司、学校）
     */
    private String tag;

    /**
     * 经度
     */
    private String longitude;

    /**
     * 纬度
     */
    private String latitude;

}

File: fast-ui/apps/admin-vue/src/api/mall/malluseraddress.ts
import request from '@/utils/axios.ts'

export interface MallUserAddressVo {
  id?: number
  userId?: number
  receiverName?: string
  receiverPhone?: string
  province?: string
  provinceCode?: string
  city?: string
  cityCode?: string
  district?: string
  districtCode?: string
  detailAddress?: string
  postalCode?: string
  isDefault?: number
  tag?: string
  longitude?: string
  latitude?: string
}

export interface MallUserAddressQuery {
  page: number
  pageSize: number
  userId?: number
  isDefault?: number
}

export interface MallUserAddressCreate {
  userId?: number
  receiverName?: string
  receiverPhone?: string
  province?: string
  provinceCode?: string
  city?: string
  cityCode?: string
  district?: string
  districtCode?: string
  detailAddress?: string
  postalCode?: string
  isDefault?: number
  tag?: string
  longitude?: string
  latitude?: string
}

export interface MallUserAddressUpdate extends MallUserAddressCreate {
  id: number
}

export interface PageVo<T> {
  data: T[]
  total: number
}

export interface ResultVo<T> {
  code: number
  data: T
  msg: string
}

export const getUserAddressPage = (data: MallUserAddressQuery) => {
  return request<ResultVo<PageVo<MallUserAddressVo>>>({
    url: '/mall/userAddress/page',
    method: 'POST',
    data,
  })
}

export const getUserAddressById = (id: number) => {
  return request<ResultVo<MallUserAddressVo>>({
    url: `/mall/userAddress/${id}`,
    method: 'GET',
  })
}

export const createUserAddress = (data: MallUserAddressCreate, requestId?: string) => {
  return request<ResultVo<object>>({
    url: '/mall/userAddress',
    method: 'POST',
    data,
    headers: { 'x-request-id': requestId }
  })
}

export const updateUserAddress = (data: MallUserAddressUpdate, requestId?: string) => {
  return request<ResultVo<object>>({
    url: '/mall/userAddress',
    method: 'PUT',
    data,
    headers: { 'x-request-id': requestId }
  })
}

export const deleteUserAddress = (id: number) => {
  return request<ResultVo<object>>({
    url: `/mall/userAddress/${id}`,
    method: 'DELETE',
  })
}

export const batchDeleteUserAddress = (ids: number[]) => {
  return request<ResultVo<object>>({
    url: '/mall/userAddress/batch',
    method: 'DELETE',
    data: ids,
  })
}

export const getUserAddressList = () => {
  return request<ResultVo<MallUserAddressVo[]>>({
    url: '/mall/userAddress/list',
    method: 'GET',
  })
}

export const getUserAddressByUser = (userId: number) => {
  return request<ResultVo<MallUserAddressVo[]>>({
    url: `/mall/userAddress/byUser/${userId}`,
    method: 'GET',
  })
}

export const getUserAddressDefault = (userId: number) => {
  return request<ResultVo<MallUserAddressVo>>({
    url: `/mall/userAddress/default/${userId}`,
    method: 'GET',
  })
}
